src/app/features/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts

```diff
@@ -31,12 +39,50 @@ export class HydrateNewVehicleRecordComponent {
     );
   }
 
+  isDirty: boolean = false;
+  isInvalid: boolean = false;
+  vehicleType?: VehicleTypes;
+
+  get customSectionForms(): Array<CustomFormGroup> {
+    if (this.summary && this.vehicleType) {
+      const commonCustomSections = [this.summary.body.form, this.summary.dimensions.form, this.summary.tyres.form, this.summary.weights.form];
+
+      switch (this.vehicleType) {
+        case VehicleTypes.PSV:
+          return [...commonCustomSections, this.summary.psvBrakes!.form];
+        case VehicleTypes.HGV:
+          return commonCustomSections;
+        case VehicleTypes.TRL:
+          return [...commonCustomSections, this.summary.trlBrakes!.form];
+        default:
+          return [];
+      }
+    } else return [];
+  }
+
+  handleFormState(): void {
+    this.vehicle$.pipe(take(1)).subscribe(data => (this.vehicleType = data?.techRecord[0].vehicleType));
+    const form = this.summary?.sections.map(section => section.form).concat(this.customSectionForms);
```

Comment:
I needed to keep the handleFormState because of the subcomponents props in the html. Ive done the other stuff though (race conditions, changed the getter)

Reply:
Nice, that's great.